battle-engine.js

```diff
@@ -1055,7 +1055,10 @@ class BattlePokemon {
 	setAbility(ability, source, effect, noForce) {
 		if (!this.hp) return false;
 		ability = this.battle.getAbility(ability);
-		let oldAbility = this.ability;
+		let oldAbility = this.ability, cantModify = {battlebond:1, powerconstruct:1, rkssystem:1, shieldsdown:1};
```

Comment:
I would not put both on the same line.

We only allow double declaration in very specific conditions, and this isn't it.